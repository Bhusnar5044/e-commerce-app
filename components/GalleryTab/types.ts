import { Image } from '@/types';

export interface GalleryTabProps {
    image: Image;
}
