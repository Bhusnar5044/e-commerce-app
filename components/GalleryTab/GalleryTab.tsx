import { memo } from 'react';

import { Tab } from '@headlessui/react';
import NextImage from 'next/image';

import { cn } from '@/utils';
import { GalleryTabProps } from './types';

export const GalleryTab: React.FC<GalleryTabProps> = memo(({ image }) => {
    return (
        <Tab className="relative flex aspect-square cursor-pointer items-center justify-center rounded-md bg-white">
            {({ selected }) => (
                <div>
                    <span className="absolute h-full w-full aspect-square inset-0 overflow-hidden rounded-md">
                        <NextImage fill src={image.url} alt="" className="object-cover object-center" />
                    </span>
                    <span
                        className={cn(
                            'absolute inset-0 rounded-md ring-2 ring-offset-2',
                            selected ? 'ring-black' : 'ring-transparent',
                        )}
                    />
                </div>
            )}
        </Tab>
    );
});

GalleryTab.displayName = 'GalleryTab';
