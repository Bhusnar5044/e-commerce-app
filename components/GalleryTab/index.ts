export { GalleryTab as default } from './GalleryTab';
