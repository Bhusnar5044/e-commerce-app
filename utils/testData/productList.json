[
    {
        "id": "1",
        "category": { "id": "1", "name": "Glasses" },
        "price": "12",
        "isFeatured": true,
        "name": "Dummy",
        "size": { "id": "1", "name": "S", "value": "12" },
        "color": { "id": "1", "name": "Gray", "value": "gray" },
        "images": [{ "id": "1", "url": "" }]
    }
]
