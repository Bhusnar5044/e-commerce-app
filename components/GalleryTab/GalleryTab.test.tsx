import { Tab } from '@headlessui/react';
import { render } from '@test-utils';
import * as ProductsMock from '@utils/testData/productList.json';
import { GalleryTab } from './GalleryTab';

describe('GalleryTab', () => {
    it('should render properly', () => {
        const { container } = render(
            <Tab.Group>
                <GalleryTab image={ProductsMock[0].thumbnail} />
            </Tab.Group>,
        );
        expect(container).toMatchSnapshot();
    });
});
