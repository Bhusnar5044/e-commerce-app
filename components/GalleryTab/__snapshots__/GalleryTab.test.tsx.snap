// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryTab should render properly 1`] = `
<div>
  <button
    aria-selected="true"
    class="relative flex aspect-square cursor-pointer items-center justify-center rounded-md bg-white"
    data-headlessui-state="selected"
    id="headlessui-tabs-tab-:r0:"
    role="tab"
    tabindex="0"
    type="button"
  >
    <div>
      <span
        class="absolute h-full w-full aspect-square inset-0 overflow-hidden rounded-md"
      >
        <img
          alt=""
          class="object-cover object-center"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.dummyjson.com%2Fdata%2Fproducts%2F1%2Fthumbnail.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </span>
      <span
        class="absolute inset-0 rounded-md ring-2 ring-offset-2 ring-black"
      />
    </div>
  </button>
</div>
`;
